### PR TITLE
Pad build numbers and prefix PRs

### PR DESCRIPTION
**Pull request by gusty**
This will create stuff like:
`MyLib.0.1.0-CI0000798.nupkg` (from master)
`MyLib.0.1.0-pr17-0000798.nupkg` (from PRs)

### CODE REVIEW

File: build.cmd [hunk 1]
```diff
@@ -1,8 +1,16 @@
-dotnet pack src/CallPolly --configuration Release -o "%CD%\bin" --version-suffix CI%1
+SETLOCAL
+SET X=%1
+SET BLD=000000%X%
+SET BLD=%BLD:~-7%
+if [%2]==[] (SET V=CI%BLD%) else (SET V=pr%2-%BLD%)
```
Comment (bartelink) [inline build.cmd]: this is looking more and more like a job for a build.proj ;)

**Comment by bartelink** [comment]
Ah, good to know - so, do you want to port it to a build.proj are are we going to :shipit: ?
(I'll copy whatever you decide to merge into the converters project)
NB I know the great is the enemy of the good, but I've been suggesting to others to clone from one of these as a way to get a good template-project; if we don't polish this sort of stuff now, we'll have ripples of manual work to fix this when we get to this (I'm guessing someone wanting to build this on non-Windows will end up doing it... - except they'll probably port this to an even worse .sh monstrosity)

**Comment by gusty** [comment]
Merging this for now. I'll keep improving the build infrastructure in separate PRs.

**Comment by gusty** [comment]
What's the failure?

I have a build.proj already. See the on in XRay.

**Comment by bartelink** [comment]
About pull request id - see the master commit log. Unfortunately the removal of your commit did not help as the build config cant cope with that (we also need to fix that - I'll leave it the way it was now)

**Comment by bartelink** [comment]
or is our build numbering predicated on the commit index on a branch? (for early days of a project I tend to do force pushes to keep history neat rather than doing formal revert PRs and tiny fixup PRs)

**Comment by bartelink** [comment]
link to error log on master https://jet-tfs.visualstudio.com/Jet/_build/results?buildId=70166&view=logs

**Comment by gusty** [comment]
Weird, I'll have a look shortly, but it seems to be a problem in the pipelines rather than the build script.

That's why removing this commit didn't solve it.

**Comment by gusty** [comment]
The problem was that Azure Pipelines was not expanded uninitialized variables.
I did a workaround in the Pipeline itself, so now it should work fine for both projects.

**Comment by bartelink** [comment]
OK, I've elided it from master to save the distration of having to analyze - #12 will presumably reinstate when it merges (I'm doing some integration work now in #11 which I will merge as soon as)

**Pull request merged**